cluster_experiments/experiment_analysis.py

```diff
@@ -680,3 +682,226 @@ def analysis_point_estimate(self, df: pd.DataFrame, verbose: bool = False) -> fl
+            df (pd.DataFrame): The dataframe containing the data after the intervention.
+            treatment_cluster (str): The name of the treatment cluster.
+
+        Returns:
+            pd.DataFrame: The dataframe with the synthetic results added to the treatment cluster.
+        """
+        weights = self.fit_synthetic(
+            pre_experiment_df=pre_experiment_df, verbose=verbose
+        )
+
+        synthetic = (
+            df[self._get_cluster_column(df) != treatment_cluster]
+            .pivot(index=self.time_col, columns=self.cluster_cols)[self.target_col]
+            .values.dot(weights)
+        )
+
+        return df[self._get_cluster_column(df) == treatment_cluster].assign(
+            synthetic=synthetic
+        )  # add synthetic to treatment cluster
+
+    def pvalue_based_on_hypothesis(self, ate: float, avg_effects: dict) -> float:
+        """
+        Returns the p-value of the analysis.
+        1. calculate the average effect after intervention for each unit.
+        2. count how many times the average effect is greater than the real treatment unit
+        3. Divide by the number of units. The result is the p-value using Fisher permutation test.
+        """
+
+        avg_effects = list(avg_effects.values())
+
+        if HypothesisEntries(self.hypothesis) == HypothesisEntries.LESS:
+            return np.mean(avg_effects < ate)
+        if HypothesisEntries(self.hypothesis) == HypothesisEntries.GREATER:
+            return np.mean(avg_effects > ate)
+        if HypothesisEntries(self.hypothesis) == HypothesisEntries.TWO_SIDED:
+            avg_effects = np.abs(avg_effects)
+            return np.mean(avg_effects > ate)
+
+        raise ValueError(f"{self.hypothesis} is not a valid HypothesisEntries")
+
+    def _get_treatment_cluster(self, df: pd.DataFrame) -> str:
+        """Returns the treatment cluster"""
+        treatment_df = df[df[self.treatment_col] == 1]
+
+        treatment_cluster = self._get_cluster_column(treatment_df).unique()[0]
+        return treatment_cluster
+
+    def analysis_pvalue(self, df: pd.DataFrame, verbose: bool = False) -> float:
+        """
+        Calculate the p-value using the treatment effect calculated from analysis_point_estimate
+        for the actual treatment cluster and comparing it with placebo effects from other clusters.
+        """
+
+        clusters = self._get_cluster_column(df).unique()
+        treatment_cluster = self._get_treatment_cluster(df)
+
+        synthetic_donors = {
+            cluster: self.analysis_point_estimate(
+                treatment_cluster=cluster,
+                df=df,
+                verbose=verbose,
+            )
+            for cluster in clusters
+        }
+
+        ate = synthetic_donors[treatment_cluster]
+        synthetic_donors.pop(treatment_cluster)
+
+        return self.pvalue_based_on_hypothesis(ate=ate, avg_effects=synthetic_donors)
+
+    def analysis_point_estimate(
```

Comment:
can you add a test for this? add a very big effect and check that the point estimate is similar